test/ptokens-router.test.js

```diff
@@ -173,6 +173,20 @@ describe('pTokens Router Contract', () => {
           }
         })
 
+      const decodeMetadataEvent = _event =>
```

Comment:
Just wondering... why wrapping sync code into a Promise considering how this function is used? Also considering the catch is rejecting with the same error.

Reply:
Just because the `codec.decode(...)` fxn call is fallible, and will `throw` on an error, but this is "hidden" from the caller (unless they actually read the function).

By wrapping it in a promise though, the user now knows the function is fallible because it's returning a promise, which could reject, and the node run time would complain about this if said rejection is unhandled.

Of course this is moot in the case of a test, but it's a good habit to make a fxns fallibility extremely clear imo.

Reply:
> By wrapping it in a promise though, the user now knows the function is fallible because it's returning a promise, which could reject, and the node run time would complain about this if said rejection is unhandled.

Makes sense